matrix.md

```diff
@@ -0,0 +1,136 @@
+# Compartmentalization Evaluation Matrix
+
+> Complete this matrix for a single implementation or system.  
+> Use ✅ / ❌ for checkboxes, categorical scales, and short text values as indicated.
+
+---
+
+## Security
+
+| **Dimension** | **Metric** | **Value / Notes** |
+|----------------|------------|-------------------|
+| **Isolation Model** | | |
+| Primary use case | Untrusted library isolation / Secret/key protection / Fault isolation / Supply chain defense / Other: ___ | |
+| Subject selection | Code-centric / Data-centric / Hybrid | |
+| Finest isolation granularity | Function / Library / Thread / Process / VM / Other: ___ | |
+| **Isolation Approach** | | |
+| Hardware primitive(s) | MPK/PKU / MTE / CHERI / TEEs / Other: ___ / None | |
+| Software technique(s) | SFI / Boundary wrappers/marshalling / Memory-safe language / Language runtime / Other: ___ / None | |
+| Isolation abstraction | Process / Thread / Protection domain / VM / Container / Other: ___ | |
+| Requires runtime software support | ✅ / ❌ (if yes, describe: ___) | |
+| **Properties Enforced** | | |
+| Security properties provided | [Describe: e.g., "Memory integrity and confidentiality, syscall filtering"] | |
+| **Resilience** | | |
+| Compartment crashes isolated | ✅ / ❌ | |
+| **Interface Safety** | | |
+| Boundaries are checked/validated | ✅ / ❌ / Partial | |
+| **Trust & Threats** | | |
+| TCB includes | Compiler / OS / Firmware / CPU / Other: ___ (list applicable) | |
+| TCB approximate size | Small / Medium / Large / Very Large or [LOC/component count if known] | |
+| Side-channel resistance | None / Partial / Strong (if Partial/Strong, describe mitigations: ___) | |
+
+---
+
+## Runtime Efficiency
+
+| **Dimension** | **Metric** | **Value / Notes** |
+|----------------|------------|-------------------|
+| General runtime overhead | [%] or Low / Medium / High | |
+| Domain switch cost | [µs] or <1µs / 1-10µs / 10-100µs / >100µs | |
+| Domain creation cost | [ms] or <1ms / 1-10ms / >10ms | |
+| Inter-domain call latency | [µs] or Low / Medium / High | |
+| Inter-domain throughput | [MB/s] or Low / Medium / High | |
```

Comment:
I am slightly unclear on what we are measuring, is that the number of bytes that can be exchanged between two domains, per second? This will be extremely dependent on memory bandwidth for many systems, I suspect that it will be hard to compare meaningfully across machines...

Reply:
right, thats probably true for any of these performance metrics. realistically we'll need to provide something of a standard spec.